libp2p/Host.cpp

```diff
@@ -942,6 +948,16 @@ void Host::restoreNetwork(bytesConstRef _b)
 	}
 }
 
+bool Host::peerSlotsAvailable(Host::PeerSlotType _type)
```

Comment:
How does this change help?

Reply:
I think I separated two lock-holdings.  And I saw a deadlock disappear.

Reply:
`peerCount()` takes a lock.

Reply:
Ah, I see, fine